### ISSUE TITLE
Loading screen / progress on first boot

### ISSUE THREAD

**Issue by philpax** [fintelia/terra#32]
Hi again! The wgpu 0.13 update seemed to fix my push constant issue, and Terra now works for me 🎉 

I had the assets from the last time I used Terra, so the initial startup this time resulted in the process becoming unresponsive with no feedback (no visual output or CLI output). As the README said this would take a while, I let it run for about ~24 hours, and came back to see that the process was still unresponsive.

I killed it and restarted it, and it worked immediately on restart, which leads me to believe that it was probably done long before I killed it.

Based on that, it might be nice to have some kind of way of telling the user how far into the initial-processing stage Terra is. The process still being unresponsive after (presumably) completing its processing might also be another bug?

Diagnostic information:
- Windows 10 21H2
- rustc 1.62.1 (e092d0b6b 2022-07-16)
- AMD Ryzen 5950HX
- nVidia GeForce RTX 3080 mobile, 16GB VRAM

---

This may also have been because I had the assets partially downloaded from a previous version of Terra, but I don't have an easy way of testing that hypothesis.

**Comment by fintelia** [comment]
Yeah, it shouldn't be taking anywhere near that long. On my machine, if I delete (or move) the cache directory then a fresh download takes under a minute, though the time will vary somewhat based on your internet speed. The note in the readme was from before I converted Terra to stream tiles from a server instead of generating them at start, which I guess was never removed.

From your description, it sounds like there was probably some sort of deadlock that happened the first time you tried running. Pausing in a debugger might reveal where that was happening
